homeassistant/components/airpatrol/climate.py

```diff
@@ -0,0 +1,321 @@
+            and any(
+                isinstance(u, dict) and u.get("unit_id") == self._unit_id
+                for u in self.coordinator.data
+            )
+            and "climate" in self._unit
+        )
+        if not is_available and not self._unavailable_logged:
+            _LOGGER.info("The climate entity '%s' is unavailable", self._attr_unique_id)
+            self._unavailable_logged = True
+        elif self._unavailable_logged:
+            _LOGGER.info("The climate entity '%s' is back online", self._attr_unique_id)
+            self._unavailable_logged = False
+        return is_available
+
+    @property
+    def current_humidity(self) -> float | None:
+        """Return the current humidity."""
+        climate_data = self._unit.get("climate", {})
+        if humidity := climate_data.get("RoomHumidity"):
+            try:
+                return float(humidity)
+            except (ValueError, TypeError):
+                _LOGGER.error("Failed to convert humidity to float: %s", humidity)
+        return None
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        climate_data = self._unit.get("climate", {})
+        if temp := climate_data.get("RoomTemp"):
+            try:
+                return float(temp)
+            except (ValueError, TypeError):
+                _LOGGER.error("Failed to convert temperature to float: %s", temp)
+        return None
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the target temperature."""
+        climate_data = self._unit.get("climate", {})
+        params = climate_data.get("ParametersData", {})
+        if temp := params.get("PumpTemp"):
+            try:
+                return float(temp)
+            except (ValueError, TypeError):
+                _LOGGER.error("Failed to convert temperature to float: %s", temp)
+        return None
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return the current HVAC mode."""
+        climate_data = self._unit.get("climate", {})
+        params = climate_data.get("ParametersData", {})
+        pump_power = params.get("PumpPower", "off")
+        pump_mode = params.get("PumpMode", "heat")
+
+        if pump_power == "off":
+            return HVACMode.OFF
+        return HVAC_MODES.get(pump_mode, HVACMode.HEAT)
+
+    @property
+    def hvac_action(self) -> HVACAction | None:
+        """Return the current HVAC action."""
+        climate_data = self._unit.get("climate", {})
+        params = climate_data.get("ParametersData", {})
+        pump_power = params.get("PumpPower", "off")
+        pump_mode = params.get("PumpMode", "heat")
+
+        if pump_power == "off":
+            return HVACAction.OFF
+
+        if pump_mode == "heat":
+            return HVACAction.HEATING
+        if pump_mode == "cool":
+            return HVACAction.COOLING
+        return HVACAction.IDLE
```

Comment:
Does this really reflect what the pump is doing? It seems you're copying what is the mode for the pump?

Reply:
Ah!